### PR TITLE
Task: Get config cmd

### LINKED ISSUE #3467: [ioctl] build config get command line into new ioctl
### What would you like to be added
 - Step 1: refactor  command  [_configGetCmd](https://github.com/iotexproject/iotex-core/blob/master/ioctl/config/configsetget.go#L42) into `./ioctl/newcmd/config/config_get.go`
 - Step 2: create `./ioctl/newcmd/config/config_get_test.go` to add unit test 

#### Notes:
1. Use [client interface](https://github.com/iotexproject/iotex-core/blob/master/ioctl/client.go) to construct the Cobra command.
2. [Output package](https://github.com/iotexproject/iotex-core/blob/master/ioctl/output) is deprecated. `e.g.` `output.PrintError` could be replaced with `errors.Wrap`.
3. Global variables (config.ReadConfig  config.DefaultConfigFile etc.) have been wrapped into client interface. Please use [client interface](https://github.com/iotexproject/iotex-core/blob/master/ioctl/client.go) to access them if needed
4. The logic of new code should be similar to [the old one](https://github.com/iotexproject/iotex-core/blob/master/ioctl/config/configsetget.go#L42).
5. In each unit test, the results should be checked whether they are as the expected. 

#### Example:
1. https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/account/accountinfo.go
2. https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/account/accountdelete.go
3. https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/account/accountcreateadd.go

### Why is this needed
Unit tests couldn't be added for the old ioctl command (./ioctl/cmd)

### PR DESCRIPTION
**Pull request by pocockn**
# Description

Step 1: refactor command [_configGetCmd](https://github.com/iotexproject/iotex-core/blob/master/ioctl/config/configsetget.go#L42) into ./ioctl/newcmd/config/config_get.go
Step 2: create ./ioctl/newcmd/config/config_get_test.go to add unit test

Fixes #3467 

## Type of change
Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] make test

### CODE REVIEW

**Comment by huof6829** [comment]
@pocockn  Thanks!

**Comment by huof6829** [comment]
Pls fix ci fail, thx.

File: ioctl/newcmd/config/config.go [hunk 1]
```diff
@@ -128,6 +129,50 @@ func (c *info) reset() error {
...
+}
+
+type endpointMessage struct {
+	Endpoint      string `json:"endpoint"`
+	SecureConnect bool   `json:"secureConnect"`
+}
+
+func (m *endpointMessage) String() string {
+	if output.Format == "" {
+		message := fmt.Sprint(m.Endpoint, "    secure connect(TLS):", m.SecureConnect)
```
Comment (huof6829) [inline ioctl/newcmd/config/config.go:170]: directly move to line142

File: ioctl/newcmd/config/config_test.go [hunk 2]
```diff
@@ -28,6 +30,100 @@ func TestInitConfig(t *testing.T) {
...
+		{
+			"all",
+			"\"endpoint\": \"http://google.com\",\n  \"secureConnect\": true,\n  \"aliases\": {},\n  \"defaultAccount\": {\n    \"addressOrAlias\": \"test\"\n  },\n  \"explorer\": \"iotexscan\",\n  \"language\": \"English\",\n  \"nsv2height\": 5165641,\n  \"analyserEndpoint\": \"https://iotex-analyser-api-mainnet.chainanalytics.org\"\n}",
+		},
+	}
+
+	for _, tc := range tcs {
+		cfgItem, err := info.get(tc.arg)
+		require.NoError(err)
+		require.Contains(cfgItem, tc.expected)
```
Comment (huof6829) [inline ioctl/newcmd/config/config_test.go:91]: expected data should be set before checking if no error

File: ioctl/newcmd/config/config_test.go [hunk 3]
```diff
@@ -28,6 +30,100 @@ func TestInitConfig(t *testing.T) {
...
+	testPath, err := os.MkdirTemp(tempDir, "testCfg")
+	require.NoError(err)
+	defer func() {
+		testutil.CleanupPath(testPath)
+	}()
+	_configDir = testPath
+	cfg, cfgFilePath, err := InitConfig()
+	info := newInfo(cfg, cfgFilePath)
+
+	tcs := []struct {
```
Comment (huof6829) [inline ioctl/newcmd/config/config_test.go:107]: combine into line50
if err!=nil {require.Contains(err.Error()  }

File: ioctl/newcmd/config/config_test.go [hunk 4]
```diff
@@ -28,6 +30,100 @@ func TestInitConfig(t *testing.T) {
...
+	}
+
+	for _, tc := range tcs {
+		cfgItem, err := info.get(tc.arg)
+		require.NoError(err)
+		require.Contains(cfgItem, tc.expected)
+	}
+}
+
+func TestConfigGetError(t *testing.T) {
```
Comment (huof6829) [inline ioctl/newcmd/config/config_test.go:95]: delete

File: ioctl/newcmd/config/config.go [hunk 5]
```diff
@@ -128,6 +129,50 @@ func (c *info) reset() error {
 	return nil
 }
 
+// get retrieves a config item from its key.
+func (c *info) get(arg string) (string, error) {
+	switch arg {
+	default:
```
Comment (huof6829) [inline ioctl/newcmd/config/config.go:135]: move to end

File: ioctl/newcmd/config/config_test.go (hunk 2 above)
Comment (pocockn) [reply ioctl/newcmd/config/config_test.go:91]: Would it be worth implementing the config set PR before this to use that functionality here?

Comment (pocockn) [reply ioctl/newcmd/config/config_test.go:87]: @huof6829  :)

**Comment by huof6829** [comment]
@pocockn   I set `endpoint=""`,  so line85 `require.Contains(err.Error(), tc.expected)`  can be checked.  `InitConfig` is done by `TestInitConfig`, so directly use test datas to check.

**Comment by pocockn** [comment]
> @pocockn I set `endpoint=""`, so line85 `require.Contains(err.Error(), tc.expected)` can be checked. `InitConfig` is done by `TestInitConfig`, so directly use test datas to check.

I see what you mean now, thank you! Are there any more changes?

File: ioctl/newcmd/config/config.go [hunk 6]
```diff
@@ -128,6 +128,42 @@ func (c *info) reset() error {
...
+	case "explorer":
+		return c.readConfig.Explorer, nil
+	case "language":
+		return c.readConfig.Language, nil
+	case "nsv2height":
+		return strconv.FormatUint(c.readConfig.Nsv2height, 10), nil
+	case "analyserEndpoint":
+		return c.readConfig.AnalyserEndpoint, nil
+	case "all":
+		return c.readConfig.String(), nil
```
Comment (huof6829) [inline ioctl/newcmd/config/config.go:156]: line146 and line156  calls output's function.  https://github.com/iotexproject/iotex-core/blob/2ed5b0387338860ed93331c19f9dc589f21da94e/ioctl/output/format.go#L180
directly refactor it here and call.  don't use output

Comment (pocockn) [reply ioctl/newcmd/config/config.go:156]: Thanks for your help, done.

File: ioctl/newcmd/config/config.go [hunk 7]
```diff
@@ -128,6 +130,42 @@ func (c *info) reset() error {
...
 
+// get retrieves a config item from its key.
+func (c *info) get(arg string) (string, error) {
+	switch arg {
+	case "endpoint":
+		if c.readConfig.Endpoint == "" {
+			return "", config.ErrEmptyEndpoint
+		}
+		message := endpointMessage{Endpoint: c.readConfig.Endpoint, SecureConnect: c.readConfig.SecureConnect}
+		return fmt.Sprint(message.Endpoint, "    secure connect(TLS):", message.SecureConnect), nil
```
Comment (huof6829) [inline ioctl/newcmd/config/config.go:141]: delete `endpointMessage`, directly use value in line141

File: ioctl/newcmd/config/config.go [hunk 8]
```diff
@@ -128,6 +130,42 @@ func (c *info) reset() error {
...
+	case "analyserEndpoint":
+		return c.readConfig.AnalyserEndpoint, nil
+	case "all":
+		return jsonString(c.readConfig), nil
+	default:
+		return "", config.ErrConfigNotMatch
+	}
+}
+
+type endpointMessage struct {
```
Comment (huof6829) [inline ioctl/newcmd/config/config.go:164]: delete

File: ioctl/newcmd/config/config.go [hunk 9]
```diff
@@ -164,3 +202,12 @@ func (c *info) loadConfig() error {
 	}
 	return nil
 }
+
+// jsonString returns json string for message
+func jsonString(out interface{}) string {
+	byteAsJSON, err := json.MarshalIndent(out, "", "  ")
+	if err != nil {
+		log.Panic(err)
```
Comment (huof6829) [inline ioctl/newcmd/config/config.go:210]: return err

**Review by huof6829 — APPROVED** [review]

**Comment by huof6829** [comment]
@pocockn  Thanks for your help!

**Review by Liuhaai — APPROVED** [review]

File: ioctl/newcmd/config/config_get.go [hunk 10]
```diff
@@ -5,3 +5,59 @@
...
+	use, _ := client.SelectTranslation(_configGetUse)
+	short, _ := client.SelectTranslation(_configGetUseCmdShorts)
+	long, _ := client.SelectTranslation(_configGetUseCmdLong)
+
+	return &cobra.Command{
+		Use:       use,
+		Short:     short,
+		Long:      long,
+		ValidArgs: _validGetArgs,
+		Args: func(cmd *cobra.Command, args []string) error {
```
Comment (Liuhaai) [inline ioctl/newcmd/config/config_get.go:46]: @huof6829  pls fix args issue

Comment (huof6829) [reply ioctl/newcmd/config/config_get.go]: fixed.  use `cobra.ExactValidArgs(1)`

**Pull request merged**